### PR TITLE
Organization roles only have a full name

### LINKED ISSUE #294: Combine name entries into a single name field for non-personal names
When the role or the name is not a personal name, switch to a single name field, not a "first name" and "last name". This will accommodate the organizations/events.   

In either case, a name entry is required.  So for personal names, the first and last name are both required.  For non personal names, the single name field is required.

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?
Fixes #294

## How was this change tested?

contributors

### CODE REVIEW

File: app/forms/work_form.rb [hunk 1]
```diff
@@ -70,15 +70,22 @@ def sync(*)
                if fragment['_destroy'] == '1'
                  contributors.delete(item)
                  return skip!
-               elsif fragment['first_name'].blank?
+               elsif fragment['first_name'].blank? && fragment['full_name'].blank?
                  return skip!
                end
+               if fragment['role_term'].start_with?('person')
```
Comment (mjgiarlo) [inline app/forms/work_form.rb:76]: Can you toss a comment here about what's going on? I grok what's going on, but I reckon we'll lose sight of this over time.

**Review by mjgiarlo — APPROVED** [review]
Toss a comment in the work form and then feel free to merge.

File: app/forms/work_form.rb (hunk 1 above)
Comment (jcoyne) [reply app/forms/work_form.rb:79]: Good call. I've added it.

**Review by mjgiarlo — APPROVED** [review]
merge when :green_circle:

**Pull request merged**